### PR TITLE
#1499: replaced instances of 'any' in AppContext.test.tsx

### LINKED ISSUE #1499: [Refactoring] - Remove `any` type from `src\frontend\src\tests\app\AppContext.test.tsx`
### Description

Remove the use of the `any` type from AppContext.test.tsx This may seem very simple, but the point of this is just to have you read through the codebase and understand the types/interfaces used in it, or make new ones that fit the shape of the data being used.

### Notes

Check #665 for notes on why we're doing this and pointers on what you should replace `any` with.

### Acceptance Criteria

`any` is not used as a type in AppContext.test.tsx

### PR DESCRIPTION
**Pull request by vsp05**
## Changes

Changed props type from "any" to other values that made more sense in the file "AppContext.test.tsx".

Closes #1499

### CODE REVIEW

File: src/frontend/src/tests/app/AppContext.test.tsx [hunk 1]
```diff
@@ -9,7 +9,9 @@ import AppContext from '../../app/AppContext';
 vi.mock('../../app/AppContextQuery', () => {
   return {
     __esModule: true,
-    default: (props: any) => {
+    default: (props: {
+      children: string | number | boolean | {} | React.ReactNodeArray | React.ReactPortal | null | undefined;
```
Comment (jackturner106) [inline src/frontend/src/tests/app/AppContext.test.tsx:13]: Okay this is interesting! I like that you found a way to make it work, but I do think we should be able to tighten up the props a little bit. Could you try making the options limited to just React.ReactNode or possibly React.ReactNode | null and see if the tests still work?

**Review by jackturner106 — COMMENTED** [review]
Good work! I think it should be possible to reduce the number of options there, I left a comment about that.

File: src/frontend/src/tests/app/AppContext.test.tsx [hunk 2]
```diff
@@ -9,7 +9,7 @@ import AppContext from '../../app/AppContext';
 vi.mock('../../app/AppContextQuery', () => {
   return {
     __esModule: true,
-    default: (props: any) => {
+    default: (props: { children: React.ReactNode | null }) => {
```
Comment (RChandler234) [inline src/frontend/src/tests/app/AppContext.test.tsx:12]: you should be able to remove the | null

**Review by RChandler234 — CHANGES_REQUESTED** [review]

**Review by jackturner106 — APPROVED** [review]
Looks good to me!

**Review by RChandler234 — APPROVED** [review]

**Pull request merged**